client/gutenberg/editor/layout/index.jsx

```diff
@@ -39,16 +56,24 @@ export class GutenlypsoLayout extends Component {
 
 export default compose( [
 	withSelect( select => {
-		const { getCurrentPost } = select( 'core/editor' );
+		const { getCurrentPost, getCurrentPostType, getEditedPostAttribute } = select( 'core/editor' );
 		return {
 			post: getCurrentPost(),
+			postType: getCurrentPostType(),
+			isTrash: 'trash' === getEditedPostAttribute( 'status' ),
 		};
 	} ),
 	withDispatch( dispatch => {
-		const { updatePost, resetBlocks } = dispatch( 'core/editor' );
+		const { editPost, savePost, updatePost, resetBlocks } = dispatch( 'core/editor' );
 		return {
+			editPost,
+			savePost,
 			updatePost,
 			resetBlocks,
 		};
 	} ),
-] )( GutenlypsoLayout );
+] )(
+	connect( ( state, { postType } ) => ( {
```

Comment:
You can include the `connect` call as the third argument of the `compose` class, which turns into more readable code.

Note that you can use both 3 arguments or a single array argument:

3 arguments:
```js
export default compose(
	withSelect(...),
	withDispatch(...),
	connect(...)
)( GutenlypsoLayout );
```
Single array argument
```js
export default compose( [
	withSelect(...),
	withDispatch(...),
	connect(...),
] )( GutenlypsoLayout );
```

Reply:
Wowee, thanks! I've never thought of that!